File: 2_class_constructor_deconstructor.cpp
#include <bits/stdc++.h>
using namespace std;

class A
{
public:
    int x;
    A()
    {
        printf("Constructor Called\n");
        x = 10;
    }
    ~A()
    {
        x = 0;
        cout << "\nDestructor Called\n"
             << x;
    }
};

int main()
{
    A myObj;
    cout << myObj.x << endl;

    return 0;
}

File: 2_class_constructor_deconstructor.py
class A:
    x = 0

    def __init__(self):
        print("Constructor Called")
        self.x = 10

    def __del__(self):
        print("Destructor Called")
        self.x = 0
        print(self.x)


myObj = A()
print(myObj.x)
